chromium_src/chrome/renderer/spellchecker/spellcheck_worditerator.cc

```diff
@@ -357,38 +364,52 @@ bool SpellcheckWordIterator::SetText(const base::char16* text, size_t length) {
   return true;
 }
 
-bool SpellcheckWordIterator::GetNextWord(base::string16* word_string,
-                                         int* word_start,
-                                         int* word_length) {
+SpellcheckWordIterator::WordIteratorStatus SpellcheckWordIterator::GetNextWord(
+    base::string16* word_string,
+    int* word_start,
+    int* word_length) {
   DCHECK(!!text_);
 
   word_string->clear();
   *word_start = 0;
   *word_length = 0;
 
   if (!text_) {
-    return false;
+    return IS_END_OF_TEXT;
   }
 
-  // Find a word that can be checked for spelling. Our rule sets filter out
-  // invalid words (e.g. numbers and characters not supported by the
-  // spellchecker language) so this ubrk_getRuleStatus() call returns
-  // UBRK_WORD_NONE when this iterator finds an invalid word. So, we skip such
-  // words until we can find a valid word or reach the end of the input string.
+  // Find a word that can be checked for spelling or a character that can be
+  // skipped over. Rather than moving past a skippable character this returns
+  // IS_SKIPPABLE and defers handling the character to the calling function.
   while (iterator_->Advance()) {
     const size_t start = iterator_->prev();
     const size_t length = iterator_->pos() - start;
-    if (iterator_->IsWord()) {
-      if (Normalize(start, length, word_string)) {
+    switch (iterator_->GetWordBreakStatus()) {
+      case base::i18n::BreakIterator::IS_WORD_BREAK: {
```

Comment:
It would be enough to handle only `base::i18n::BreakIterator::IS_WORD_BREAK` if we were to maintain consistency with existing logic. Currently we don't seem to handle `IS_SKIPPABLE_WORD`

Reply:
I am planning to come up with another PR to utilize skippable word enum status. (And would like to leave worditerator as is to chromium's)

Reply:
In that case, we should let the loop https://github.com/electron/electron/pull/11819/files#diff-bfd35a1e98d8d7cb08cc06502b38b7d8R125 continue for `IS_SKIPPABLE_WORD` until its utilized.

Reply:
make sense, will try to update.

Reply:
Yes it does iterate, but also does spell checking with `IS_SKIPPABLE_WORD`. We should instead continue the loop.

```
for (auto status = text_iterator_.GetNextWord(&word, &word_start, &word_length);
     status != SpellcheckWordIterator::IS_END_OF_TEXT;
     status = text_iterator_.GetNextWord(&word, &word_start, &word_length)) {
   if (status == SpellcheckWordIterator::IS_SKIPPABLE)
     continue;
```

Reply:
so my original intention was leave it as-is (to not to change spellcheck side behavior yet, cause current spellchecker did similar anway to check skippable word but iterator just didn't distinguish it), but sounds fair and updated per suggestion.